### PR TITLE
Fixes #33757 - reports button link to new reports

### PR DESCRIPTION
**Pull request by lzap**
Reports are changing to Host Reports:

https://community.theforeman.org/t/host-reports-plugin-in-foreman-3-1/25796

Let's advertise this on the old page and allow people to migrate to the new endpoint and format.

(Edit: Previously this patch was also about redirecting Reports button on the host detail page, but I changed my mind. We can do this via Deface if we decide it is too confusing to have two Reports button on the page.)

### CODE REVIEW

**Comment by ofedoren** [comment]
If we're deprecating/changing things in core, could you please as part of this PR also add a warning on the Config Reports page? Just paste the following snippet to the `app/views/config_reports/index.html.erb`:
```erb
<% if defined?(ForemanHostReports) %>
  <%= alert class: 'alert-warning',
    header: _('This page is going to be removed'),
    text: "</br>".html_safe +
          _("Go to Monitor > Host Reports for up-to-date information.") +
          "</br>".html_safe +
          _("Please, read Upgrade notes to migrate to the new format.") %>
<% end %>
```
Of course, please modify the message to be more appropriate.

File: app/views/config_reports/index.html.erb [hunk 1]
```diff
@@ -14,6 +14,14 @@
...
   ]
 ) if @host %>
+<% if defined?(ForemanHostReports) %>
+  <%= alert class: 'alert-warning',
+    header: _('This page is going to be removed'),
+    text: "</br>".html_safe +
+          _("Go to Monitor > Host Reports for up-to-date information.") +
+          "</br>".html_safe +
+          _("Please, read Upgrade notes to migrate to the new format.") %>
+<% end %>
```
Comment (lzap) [inline app/views/config_reports/index.html.erb]: Added. Question: should we only show this when the plugin is installed OR to all of users to advertise this feature a bit more?

**Comment by lzap** [comment]
I have repurposed this patch. This is solely just a warning now. I think we can always show it, not just when Foreman Host Reports plugin is installed. This will help to spread the news more. We want users to migrate during or after upgrade to 3.1.

**Comment by Ron-Lavi** [comment]
JS test error not related, being fixed in https://github.com/theforeman/foreman/pull/8863

File: app/views/config_reports/index.html.erb [hunk 2]
```diff
@@ -14,6 +14,12 @@
     },
   ]
 ) if @host %>
+<%= alert class: 'alert-warning',
+  header: _('This page is going to be removed'),
+  text: "</br>".html_safe +
+        _("Go to Monitor > Host Reports for up-to-date information.") +
```
Comment (tbrisker) [inline app/views/config_reports/index.html.erb]: Will this page be there if the plugin isn't installed?

**Review by tbrisker — CHANGES_REQUESTED** [review]
It would be good to add a link to somewhere explaining why this page will be removed and what will be replacing it and when, e.g. the discourse post on the subject.

File: app/views/config_reports/index.html.erb (hunk 2 above)
Comment (ofedoren) [reply app/views/config_reports/index.html.erb]: No, and that's why I suggested to wrap it with:
```ruby
<% if defined?(ForemanHostReports) %>
...
<% end %>
```

Comment (tbrisker) [reply app/views/config_reports/index.html.erb]: We could just add the alert from the plugin using deface in that case, while deface is not a good permanent fix it could be the easiest for adding something that will only be there for a couple of versions.

Comment (lzap) [reply app/views/config_reports/index.html.erb]: Right, let me rephrase this completely.

**Comment by lzap** [comment]
Yeah, I created a forum placeholder where I am going to put migration and installation instructions: https://community.theforeman.org/t/migrating-reports-to-the-new-format-in-foreman-3-1/25846

Rebased.

**Comment by lzap** [comment]
Fixed the HTML link, this is now ready, @tbrisker can we hit 3.1RC1 with this one? This is zero risk, we can merge afterwards.

**Review by ofedoren — DISMISSED** [review]
From my perspective this is OK now. The only thing is a weird commit.

**Comment by lzap** [comment]
Damn! Fixed.

**Review by tbrisker — APPROVED** [review]
Thanks @lzap ! will fix commit message on merge (it still talks about the button...)
